src/prim/unix/prim.c

```diff
@@ -430,13 +430,29 @@ int _mi_prim_commit(void* start, size_t size, bool* is_zero) {
     err = errno;
     unix_mprotect_hint(err);
   }
+
+  #if defined(__APPLE__)
+    // MADV_FREE_REUSABLE is paired with MADV_FREE_REUSE for accounting
+    // if this memory was not marked as MADV_FREE_REUSABLE, this call is noop
+    madvise(start, size, MADV_FREE_REUSE);
+  #endif
   return err;
 }
 
 int _mi_prim_decommit(void* start, size_t size, bool* needs_recommit) {
   int err = 0;
-  // decommit: use MADV_DONTNEED as it decreases rss immediately (unlike MADV_FREE)
-  err = unix_madvise(start, size, MADV_DONTNEED);
+  #if defined(__APPLE__)
+    // On Apple platforms there is MADV_FREE_REUSABLE -- it marks memory as RESUABLE to reduce memory pressure.
+    err = unix_madvise(start, size, MADV_FREE_REUSABLE);
+    // if this fails - fallback to MADV_DONTNEED
+  #endif
+
+  if (err)
```

Comment:
On non-Apple platforms, `err` is still 0 here, right? Or am I misreading?

Reply:
yes, you're right! I made a mistake, sorry!

Reply:
fixed, could you please look again?